src/python/dxpy/scripts/dx.py

```diff
@@ -5051,6 +5055,8 @@ def register_parser(parser, subparsers_action=None, categories=('other', ), add_
 parser_new_project.add_argument('--bill-to', help='ID of the user or org to which the project will be billed. The default value is the billTo of the requesting user.')
 parser_new_project.add_argument('--phi', help='Add PHI protection to project', default=False,
                                 action='store_true')
+parser_new_project.add_argument('--database-ui-view-only', help='Restrict viewers on the project from accessing the database details directly', default=False,
```

Comment:
I don't think we need to add a support for `databaseUIViewOnly` flag also to project new. I would treat it in a same way as `restricted` and `downloadRestricted` flags as these flags can be updated. `containsPHI` flag can't be changed once it is set on a project because it's tight to PHI access restrictions so therefore it makes sense to have it here.